File: src/main/java/org/radarcns/hdfs/data/ZipCompression.java
package org.radarcns.hdfs.data;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.util.Collection;
import java.util.Collections;
import java.util.zip.ZipEntry;
import java.util.zip.ZipInputStream;
import java.util.zip.ZipOutputStream;

public class ZipCompression implements Compression {
    @Override
    public OutputStream compress(String name, OutputStream out) throws IOException {
        ZipOutputStream zipOut = new ZipOutputStream(out);
        zipOut.setLevel(7);
        zipOut.putNextEntry(new ZipEntry(name));
        return zipOut;
    }

    @Override
    public InputStream decompress(InputStream in) throws IOException {
        ZipInputStream zipIn = new ZipInputStream(in);
        zipIn.getNextEntry();
        return zipIn;
    }

    @Override
    public Collection<String> getFormats() {
        return Collections.singleton("zip");
    }

    @Override
    public String getExtension() {
        return ".zip";
    }
}

File: src/main/java/org/radarbase/output/compression/ZipCompression.kt
package org.radarbase.output.compression

import java.io.IOException
import java.io.InputStream
import java.io.OutputStream
import java.util.zip.ZipEntry
import java.util.zip.ZipInputStream
import java.util.zip.ZipOutputStream

class ZipCompression : Compression {

    override val formats = setOf("zip")

    override val extension = ".zip"

    @Throws(IOException::class)
    override fun compress(fileName: String, out: OutputStream) = ZipOutputStream(out).apply {
        setLevel(7)
        putNextEntry(ZipEntry(fileName))
    }

    @Throws(IOException::class)
    override fun decompress(`in`: InputStream): InputStream = ZipInputStream(`in`).apply {
        nextEntry
    }
}
